src/layout/mod.rs

```diff
@@ -82,13 +83,13 @@ pub fn graph_spring_layout(
     pos: Option<HashMap<usize, Point>>,
     fixed: Option<HashSet<usize>>,
     k: Option<f64>,
-    repulsive_exponent: Option<i32>,
-    adaptive_cooling: Option<bool>,
-    num_iter: Option<usize>,
-    tol: Option<f64>,
+    repulsive_exponent: i32,
+    adaptive_cooling: bool,
+    num_iter: usize,
+    tol: f64,
     weight_fn: Option<PyObject>,
     default_weight: f64,
-    scale: Option<f64>,
+    scale: f64,
```

Comment:
Is this a consequence of https://github.com/PyO3/pyo3/pull/2703?

Reply:
It's related to the signature changes more generally, but I don't think because of that pr directly (the changes to the visitor function were directly because of that pr). The changes here were because of the type mismatch between the default values set in the signature and these types, `2. != Option<f64>`

Reply:
I guess we could have used `Some(2.0)` instead. But because 99% call `spring_layout` from our `__init__.py` which never provides `None` to `graph_spring_layout`, I think this is fine